api/src/main/java/com/microsoft/gctoolkit/time/DateTimeStamp.java

```diff
@@ -284,26 +248,20 @@ public DateTimeStamp add(double offsetInDecimalSeconds) {
         if (Double.isNaN(offsetInDecimalSeconds))
             throw new IllegalArgumentException("Cannot add " + Double.NaN);
 
-        DateTimeStamp now;
         double adjustedTimeStamp = Double.NaN;
-        if ( ! Double.isNaN(getTimeStamp())) {
+        ZonedDateTime adjustedDateStamp = null;
+        if ( hasTimeStamp()) {
             adjustedTimeStamp = getTimeStamp() + offsetInDecimalSeconds;
         }
 
-        // Need to scale the values from ZonedDateTime
-        if (getDateTime() != null) {
+        if (hasDateStamp()) {
             double offset = (Double.isNaN(offsetInDecimalSeconds)) ? 0.000d : offsetInDecimalSeconds;
             int seconds = (int) offset;
             long nanos = ((long) ((offset % 1) * 1_000L)) * 1_000_000L;
-            ZonedDateTime adjustedDateStamp = dateTime.plusSeconds(seconds).plusNanos(nanos);
-            if ( Double.isNaN(getTimeStamp()))
-                now = new DateTimeStamp(adjustedDateStamp);
-            else
-                now = new DateTimeStamp(adjustedDateStamp, adjustedTimeStamp);
-        } else
-            now = new DateTimeStamp(adjustedTimeStamp);
+            adjustedDateStamp = dateTime.plusSeconds(seconds).plusNanos(nanos);
+        }
 
-        return now;
+        return new DateTimeStamp(adjustedDateStamp, adjustedTimeStamp);
```

Comment:
This doesn't seem right. I think you may be counting the offset twice under certain circumstances. If the DateTimeStamp has both date and time stamps, then you will have adjusted the timeStamp by the offset and the dateStamp by the offset.

Reply:
Yes, the date stamp will be adjusted along with the date stamp. Maybe more documentation is needed here.
The time stamp is the age of the JVM (unless not set). Date stamp is the calendar time. Both need to be adjust to keep them coherent.